### PR TITLE
Blending mode enum

### PR DESCRIPTION
**Pull request by tischi**
Hi @NicoKiaru,

It looks like it's working.

At some point we could also make the Projector class an enum.

Let me know if you feel we should do this also in this PR.

### CODE REVIEW

**Comment by NicoKiaru** [comment]
OK, nice, did you run the tests ? I anticipate that the way the blending mode is stored into the xml file could have changed (which is not a big deal for the moment).

**Comment by tischi** [comment]
> OK, nice, did you run the tests ? 

I did `mvn clean install` does that run the tests for you? Or do you have a different mechanism?

**Comment by NicoKiaru** [comment]
In IntelliJ you can do `run all tests` and know which pass and which does not pass

**Comment by tischi** [comment]
> In IntelliJ you can do run all tests and know which pass and which does not pass

No maven way of doing this?

**Comment by tischi** [comment]
I can't find the `run all test` in intellij

**Comment by tischi** [comment]
@NicoKiaru I still don't manage to run the test. In my IntelliJ I only have this when I right-click on the test folder:

image

And that wants to run python tests for me... ☹️ 

I also tried with `mvn` but got weird errors.

Do you have any idea?

**Comment by NicoKiaru** [comment]
Weird. I got this:

image

Did you try to update intellij ? Which version do you have ? I have:

IntelliJ IDEA 2020.3 (Community Edition)
Build #IC-203.5981.155, built on November 30, 2020
Runtime version: 11.0.9+11-b1145.21 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 1957M
Cores: 8
Registry: debugger.watches.in.variables=false

**Comment by NicoKiaru** [comment]
If the tests run on mvn clean install it's probably fine, but it's good to be able to run all tests independently because you have better 'reports'

**Comment by tischi** [comment]
> If the tests run on mvn clean install it's probably fine, but it's good to be able to run all tests independently because you have better 'reports'

They don't run for me on `mvn clean install` in the playground. I think there's a `mvn` plugin missing. I will investigate the IntelliJ version.

**Comment by tischi** [comment]
I downloaded the same version of IntelliJ as you have and the menu entry is still not there.
I have no idea. I googled a bit and around 2013 other people seemed to have similar issues, which were hard to fix.

Is it OK if I try to modify the pom such that we can run tests with maven?

**Comment by tischi** [comment]
@NicoKiaru I managed to enable `mvn test`, the trick was to both add a mvn plugin and rename the `src/test/src` folder to `src/test/java`. I hope that's OK for you?

Now when running `mvn test` I got an error that I do not understand (due to lack of scijava knowledge):

```
demoRunOk(sc.fiji.bdvpg.io.PerfOpenMultipleSpimDataTest)  Time elapsed: 1.607 sec  <<< ERROR!
java.lang.IllegalArgumentException: Invalid service: sc.fiji.bdvpg.scijava.services.SourceAndConverterBdvDisplayService
        at sc.fiji.bdvpg.io.PerfOpenMultipleSpimDataTest.main(PerfOpenMultipleSpimDataTest.java:69)
        at sc.fiji.bdvpg.io.PerfOpenMultipleSpimDataTest.demoRunOk(PerfOpenMultipleSpimDataTest.java:103)
```

Do you have an idea what this could be due to?

**Comment by NicoKiaru** [comment]
It's fine for me but not for travis, because the tests require the UI. (with your modifications the tests are run on travis, while I was only running them locally). So the build is failing now, which should not happen.

Can you at least run the tests individually in the IDE, if not in batch ?

We can have a short zoom if you wish

**Comment by tischi** [comment]
@NicoKiaru I managed to add the tests. The JUnit plugin was missing in the IntelliJ plugins!
But, I am still getting above error:
```
java.lang.IllegalArgumentException: Invalid service: sc.fiji.bdvpg.scijava.services.SourceAndConverterBdvDisplayService
        at sc.fiji.bdvpg.io.PerfOpenMultipleSpimDataTest.main(PerfOpenMultipleSpimDataTest.java:69)
        at sc.fiji.bdvpg.io.PerfOpenMultipleSpimDataTest.demoRunOk(PerfOpenMultipleSpimDataTest.java:103)
```
Could you maybe pull the branch and see whether you can reproduce?

**Comment by NicoKiaru** [comment]
Great news! Can you run mvn clean install, re run the test and see if you still have the issue ? Sometimes the Services are not detected after a first build. If it doesn't work, I'll check the branch

**Comment by tischi** [comment]
I tried. Does not work. It is very weird: when I run the main methods it is working; but it is not working when I run the test methods, which should be identical the way I understand your code.

**Comment by NicoKiaru** [comment]
Ok, for me it works on your branch, running it from main or test. Weird but not a big issue.
I'll do a quick review, but we should be able to merge.

File: src/main/java/sc/fiji/bdvpg/bdv/ScreenShotMaker.java [hunk 1]
```diff
@@ -52,10 +52,7 @@
 import net.imglib2.util.Intervals;
 import net.imglib2.view.IntervalView;
 import net.imglib2.view.Views;
-import sc.fiji.bdvpg.bdv.projector.AccumulateAverageProjectorARGB;
-import sc.fiji.bdvpg.bdv.projector.AccumulateMixedProjectorARGB;
-import sc.fiji.bdvpg.bdv.projector.AccumulateSumProjectorARGB;
-import sc.fiji.bdvpg.bdv.projector.Projection;
+import sc.fiji.bdvpg.bdv.projector.*;
```
Comment (NicoKiaru) [inline src/main/java/sc/fiji/bdvpg/bdv/ScreenShotMaker.java]: We should avoid wildcard import. I know intellij automatically puts them, but that should be removed

**Review by NicoKiaru — APPROVED** [review]

File: src/main/java/sc/fiji/bdvpg/bdv/ScreenShotMaker.java (hunk 1 above)
Comment (tischi) [reply src/main/java/sc/fiji/bdvpg/bdv/ScreenShotMaker.java]: Done

**Comment by NicoKiaru** [comment]
oki doc, merge?

**Pull request merged**